### ISSUE TITLE
enh: accessibility

### ISSUE THREAD

**Issue by tjbck** [open-webui/open-webui#2790]
#277

**Comment by evolu8** [comment]
Trying to understand why the WCAG 2.1 request was closed in favour of this less detailed issue?  Either way I'd like to highlight the value of the #277 thread to readers of this.

**Comment by devinprater** [comment]
The latest update from a few days ago fixed a lot of buttons. Now, the only two issues I have are:

* In the model list, in a new chat, where I choose a model, the models are just read as "model-item" instead of the model name.
* When models have generated a response, I expect my screen reader to read out the response, similar to how ChatGPT does it as of a week ago or so. This can be done using [Live Regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions). Other than that, as far as I've tried today, it's pretty well improved! Thank you very much!

**Comment by vojtapolasek** [comment]
I was pointed to this issue. I started experimenting with this software few days ago. @devinprater  how about buttons which are just above the chat edit field? Where you can do thumbs up / down, stop generation etc. ARe they accessible for you? For me they are not, so maybe I need some later version of the software.

**Comment by devinprater** [comment]
No, they are not. They've not been labeled yet.
Devin Prater
***@***.***

**Comment by vojtapolasek** [comment]
@devinprater  One more thing to confirm... when on Linux and using Orca, can you really go through the list of models? When I activate the button, it expands and I am put into a search box. But when I type something, I can't use arrows to go through models. I know that models are there, but I have to use a special tool (ocrdesktop) to select the model.

**Comment by devinprater** [comment]
You'll need to use a Chromium browser, and use object navigation and go
into each "model item" to see which model you're on. It's rather complex,
but that's what I do for now.
Devin Prater
***@***.***

**Comment by aamir51** [comment]
Hello
Issue with unlabelled buttons is still present in the latest version

**Comment by silentoplayz** [comment]
Related - https://github.com/open-webui/open-webui/issues/18420

**Comment by ncecere** [comment]
To support adoption by federally funded educational institutions and other public sector organizations, I was wondering if there has been any movement to make sure that Open WebUI meets WCAG 2.2.   

The reason I ask is because public institution's running Open WebUI could face fines if it doesn't. I'm not sure if any of the enterprise customers have brought this but it will be required by April 24th for a lot of organizations. A lot of other projects and software I have been working with have just been publishing there WCVAG 2.2 report card on there documentation site so that we can reference that. That is all I was trying to bring up is that if the Open WebUI had any accessibility reports , it would be nice if they were available on the documentation site.

**Comment by Classic298** [comment]
@ncecere i will take this to the team but if you require such a report your best bet is to contact open webui sales for enterprise.

**Comment by tjbck** [comment]
#27601

**Comment by TomTheWise** [comment]
@tjbck Hi tim, while I appreciate the efforts and improvements for the now called accessability mode, sadly 0.11.X is still a big regression / set back on readability and look and feel - especially in normal mode.

- Bold text is still nearly indistinguishable from normal text as it is not really bold. 
   - Fix would be simply +100 font-weight for strong text (should be applied in BOTH modes)
- 80% of UI doesn't correctly hover over, some seem to hover over with only a tiniest change in bg color, some none, only a few hover over / change bg correctly.
   - Pinned Chats for example still hover overs correctly - this should be (re?-)applied to other stuff (should be applied in BOTH modes)
- Modelnames have nearly same font styling as normal text - so a model that does not reason is looking like the first part of the actual chat answer.
   - Fix Make model names font styling in chat distinguishable like a bit bolder or larger (should be applied in BOTH modes)


## Hover overs correctly examples:

<img width="513" height="178" alt="Image" src="https://github.com/user-attachments/assets/258e0f67-d084-4291-82a3-e965e6c14d2f" />

<img width="626" height="310" alt="Image" src="https://github.com/user-attachments/assets/83a9f12f-63da-426b-b3c4-16118fd29ceb" />


## Hover over not correctly examples:

<img width="427" height="244" alt="Image" src="https://github.com/user-attachments/assets/49caea0f-82b2-41b2-a1bc-0502cd02d4f4" />

<img width="393" height="100" alt="Image" src="https://github.com/user-attachments/assets/37ff63e0-4424-4543-9830-891404c3b6d6" />

<img width="325" height="170" alt="Image" src="https://github.com/user-attachments/assets/35317b9d-7471-4c60-9903-aa1a0c8773d7" />

<img width="485" height="176" alt="Image" src="https://github.com/user-attachments/assets/959b4781-4eac-4eef-b768-d3d5868760d7" />


## Personal opinion

IMHO normal mode should already be very much WCAG compliant and good readable!
Meanwhile an Accessability mode - like on smartphones, apps or websites that feature a special accessabily mode - should be super accessable and help those who need it to use the device/app/site.
